### PR TITLE
[camera_windows] Use `CameraAccessDenied` error code for camera permission errors

### PR DESCRIPTION
**Pull request by loic-sharma**
## Background

The camera plugin reports issues to the Flutter app by throwing [`CameraException`](https://pub.dev/documentation/camera_platform_interface/latest/camera_platform_interface/CameraException-class.html)s. As per [this proposal](https://docs.google.com/document/d/1t_5FYzOTlBWC6arHc4_8VuG0BUOtIyP9xWvbDER5Y9I/edit), we would like to standardize the `CameraException`'s `code` value so that common errors can be handled consistently across platforms.

This change makes the Windows camera plugin set the error code to `CameraAccessDenied` when camera permissions were denied. This is a breaking change as previously the error code would've been `camera_error` or `capture_error`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_windows/CHANGELOG.md
Comment (loic-sharma) [inline packages/camera/camera_windows/CHANGELOG.md:5]: This may not be clear, but a camera permission error during capture will still have a code of `CameraAccessDenied`. Let me know if you have suggestions on how to word this better.

File: packages/camera/camera_windows/windows/camera.cpp
Comment (loic-sharma) [inline packages/camera/camera_windows/windows/camera.cpp:253]: This is one of the breaking changes. This case can be triggered if I unplug my camera while the preview us running. It's not clear why this case had its own error code, it seems better for these error codes to be consistent.

**Review by yaakovschectman — COMMENTED** [review]
This is probably just something I need to learn about as I am new here, my only question is why there are two checks listed as failing (publishable and submit-queue), or what their failure indicates?

**Comment by loic-sharma** [comment]
@yaakovschectman Good question! You can pop open the `Details` link next to each failure to learn more about what happened:

* `publishable` - Looks like my change is using an invalid version number. I'll need to address this before merging this pull request. That's my mistake.
* `submit-queue` - We want the `main` branch to build and pass tests at all times. Whenever this isn't the case, we should stop merging new things to `main` until the branch is fixed. Common failures on `main` branch include flaky tests, or, test failures from post-submit tests (these are tests that aren't run on pull requests until after they have been merged). If you haven't already, I recommend reading through this: https://github.com/flutter/flutter/wiki/Tree-hygiene

File: packages/camera/camera_windows/windows/camera.cpp
Comment (cbracken) [inline packages/camera/camera_windows/windows/camera.cpp:33]: Replace the default with a fallthrough with both remaining cases listed.

We should probably add a lint that bans `default` cases like we do elsewhere, so that we're forced to deal with each usage site if new enum values are ever added.

File: packages/camera/camera_windows/windows/test/camera_test.cpp
Comment (cbracken) [inline packages/camera/camera_windows/windows/test/camera_test.cpp:156]: uber-nitpick: could be const.

File: packages/camera/camera_windows/windows/test/capture_controller_test.cpp
Comment (cbracken) [inline packages/camera/camera_windows/windows/test/capture_controller_test.cpp:395]: uber-pendantic-nit: coust be const/constexpr.

**Review by cbracken — APPROVED** [review]
lgtm modulo nits! Error handling is a lot clearer now! Thanks!

File: packages/camera/camera_windows/windows/test/camera_test.cpp
Comment (loic-sharma) [reply packages/camera/camera_windows/windows/test/camera_test.cpp]: Fixed!

File: packages/camera/camera_windows/windows/test/capture_controller_test.cpp
Comment (loic-sharma) [reply packages/camera/camera_windows/windows/test/capture_controller_test.cpp:395]: Fixed!

File: packages/camera/camera_windows/windows/camera.cpp
Comment (loic-sharma) [reply packages/camera/camera_windows/windows/camera.cpp:34]: I added all the possible cases, however, I wasn't able to remove the `default` case as that resulted in this error:

```
... error C2220: the following warning is treated as an error ...
... warning C4715: 'camera_windows::GetErrorCode': not all control paths return a value ...
camera_windows/example unit tests failed to build.
```

I moved the `default` case to be the fall-through after all other cases.

File: packages/camera/camera_windows/windows/camera.cpp
Comment (hellohuanlin) [inline packages/camera/camera_windows/windows/camera.cpp:21]: maybe `camera_error_others`?

**Review by hellohuanlin — COMMENTED** [review]
Thanks for making this change. In the future we should think about further clean up all errors across platforms, and for other plugins as well.

File: packages/camera/camera_windows/windows/camera.cpp
Comment (loic-sharma) [reply packages/camera/camera_windows/windows/camera.cpp:21]: The existing plugin already uses `camera_error` as its fallback error code. Changing this to `camera_error_others` would be a breaking change, which I'd rather do in a separate pull request since this one focuses on introducing camera permission errors.

Also, if we were to do this breaking change I'd prefer to go to a standardized fallback error code. None exists today, but perhaps `CameraError` would be better aligned to the other standardized camera error code?

**Pull request merged**
